### PR TITLE
Relationships + field util to comply with jsonapi 1.0

### PR DESCRIPTION
**Pull request by JliptonRTR**

### CODE REVIEW

File: dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/util/FieldUtil.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package io.rtr.jsonapi.util;
+
+import com.google.common.collect.Lists;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+
+public class FieldUtil {
```
Comment (zain-cheng) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/util/FieldUtil.java:9]: Unit tests... Might as well start now.  http://sonarqube01.prod.renttherunway.it:9000/drilldown/measures/3805?metric=coverage

**Comment by zain-cheng** [comment]
Just that one comment, but lgtm

File: dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/util/FieldUtil.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  public static Field findDeclaredField(Object obj, String fieldName) throws NoSuchFieldException {
+    List<Field> fields = Lists.newArrayList();
+    getAllFields(fields, obj.getClass());
+    for(Field field : fields) {
+      if(field.getName().equals(fieldName)) {
+        return field;
+      }
+    }
+    throw new NoSuchFieldException("No Field found for fieldName=" + fieldName + ", Obj=" + obj.getClass().getCanonicalName());
+  }
```
Comment (jadeglaze) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/util/FieldUtil.java:30]: try {
return fields.stream().filter(f -> f.getName().equals(fieldName)).findFirst().get()
}
catch (NoSuchElementException e) {
throw new NoSuchFieldException..... blah blah...
}

Comment (JliptonRTR) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/util/FieldUtil.java:30]: Idk, that's about 20 more characters than the loop version

File: dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/util/FieldUtil.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  /**
+   * A recursive method to find all the fields (including fields of the class's parents) of a class
+   *
+   * @param fields A container for Field objects, which is used by the recursion
+   * @param type The class that the recursion starts with
+   * @return a list of fields from the given class and all of its parents
+   */
+  public static List<Field> getAllFields(List<Field> fields, Class<?> type) {
```
Comment (jadeglaze) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/util/FieldUtil.java:39]: There should be another non-recursive version that hides the recursion for callers. Also, LinkedList might be better here... Or even HashSet?

getAllFields(Class<?> type) {
return getAllFields(Lists.newLinkedList(), type);
}

File: dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/util/FieldUtil.java (hunk 2 above)
Comment (jadeglaze) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/util/FieldUtil.java:30]: :)

**Pull request merged**